app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/TextInputViewModel.kt

```diff
@@ -121,4 +150,32 @@ class TextInputViewModel private constructor(
       )
     }
   }
+
+  private enum class TextParsingUiError(@StringRes private var error: Int?) {
+    /** Corresponds to non empty input. */
+    VALID(error = null),
+
+    /** Corresponds to empty input. */
+    EMPTY_INPUT(error = R.string.text_error_empty_input);
+
+    /**
+     * Returns the string corresponding to this error's string resources, or null if there is none.
+     */
+    fun getErrorMessageFromStringRes(resourceHandler: AppLanguageResourceHandler): String? =
+      error?.let(resourceHandler::getStringInLocale)
+
+    companion object {
+      /**
+       * Returns the [TextParsingUiError] corresponding to the input.
+       */
+      fun createFromParsingError(text: String): TextParsingUiError {
+
+        return if (text.isEmpty()) {
+          EMPTY_INPUT
+        } else {
+          VALID
+        }
+      }
```

Comment:
```suggestion
      /** Returns the [TextParsingUiError] corresponding to the input. */
      fun createForText(text: String): TextParsingUiError =
        EMPTY_INPUT if (text.isEmpty()) else VALID
```

Slight nit: can be done very compactly.

Also, I renamed the method to be a bit more clear on what it's doing.

Reply:
OKay.